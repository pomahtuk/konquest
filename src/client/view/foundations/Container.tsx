import React, { ReactElement, useContext, ReactNode, HTMLAttributes } from "react";
import styled from "@emotion/styled";
import { ThemeContext } from "../themes/ThemeProvider";

export type ContainerProps = {
  children: ReactNode;
  centered?: boolean;
  className?: string;
  attributes?: HTMLAttributes<HTMLDivElement>;
};

const Container = ({ children, centered, className, attributes }: ContainerProps): ReactElement => {
  const theme = useContext(ThemeContext);
  const ContainerStyled = styled.div({
    boxSizing: "border-box",
    position: "relative",
    width: "100%",
    padding: theme.units.medium,
    ...(centered
      ? {
          margin: "0 auto",
          maxWidth: "1140px;"
        }
      : {})
  });
  return (
    <ContainerStyled className={className} {...attributes}>
      {children}
    </ContainerStyled>
  );
};

export default Container;
