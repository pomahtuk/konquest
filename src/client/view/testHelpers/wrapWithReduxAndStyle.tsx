import React, { ReactElement } from "react";
import { Provider } from "react-redux";
import storeCreator from "../../stores/game.store";
import ThemeProvider from "../themes/ThemeProvider";
import baseTheme from "../themes/base.theme";
const store = storeCreator();

export const CurrentStore = store;

const wrapWithReduxAndStyle = (element: ReactElement): ReactElement => {
  return (
    <Provider store={store}>
      <ThemeProvider theme={baseTheme}>{element}</ThemeProvider>
    </Provider>
  );
};

export default wrapWithReduxAndStyle;
