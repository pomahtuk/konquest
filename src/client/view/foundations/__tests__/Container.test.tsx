import React from "react";
import { render } from "@testing-library/react";
import { matchers } from "jest-emotion";

expect.extend(matchers);

import wrapWithReduxAndStyle from "../../testHelpers/wrapWithReduxAndStyle";
import Container from "../Container";

const containerText = "hello";

describe("<Container />", (): void => {
  it("renders <Container /> component", (): void => {
    const { container } = render(wrapWithReduxAndStyle(<Container>{containerText}</Container>));
    expect(container).toMatchSnapshot();
  });

  it("renders <Container /> as centered", (): void => {
    const { getByText } = render(wrapWithReduxAndStyle(<Container centered>{containerText}</Container>));
    const element = getByText(containerText);
    expect(element).toHaveStyleRule("margin", "0 auto");
  });

  it("renders <Container /> with additional className", (): void => {
    const { getByText } = render(wrapWithReduxAndStyle(<Container className="some">{containerText}</Container>));
    const element = getByText(containerText);
    expect(element.classList.contains("some")).toBe(true);
  });
});
