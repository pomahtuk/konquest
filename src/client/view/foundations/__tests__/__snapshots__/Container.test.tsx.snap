// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> renders <Container /> component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  position: relative;
  width: 100%;
  padding: 8px;
}

<div>
  <div
    class="emotion-0"
  >
    hello
  </div>
</div>
`;
